services/analytics.js

```diff
@@ -15,9 +15,10 @@ const fetchActions = async (collection, spaceIds, { sampleSize } = {}) => {
 
   if (sampleSize) {
     aggregateQuery.push({ $sample: { size: sampleSize } });
+    return await collection.aggregate(aggregateQuery).toArray();
```

Comment:
Can you do the `toArray()` in the function that calls this function? That way this function always returns a cursor and you don't have to worry about inconsistencies.